uaclient/contract_data_types.py

```diff
@@ -0,0 +1,609 @@
+
+
+class ExternalID(DataObject):
+    fields = [
+        Field("origin", StringDataValue, False),
+        Field("IDs", data_list(StringDataValue), False),
+    ]
+
+    def __init__(self, origin: Optional[str], IDs: Optional[List[str]]):
+        self.origin = origin
+        self.IDs = IDs
+
+
+class AccountInfo(DataObject):
+    fields = [
+        Field("name", StringDataValue, False),
+        Field("id", StringDataValue, False),
+        Field("createdAt", StringDataValue, False),
+        Field("type", StringDataValue, False),
+        Field("userRoleOnAccount", StringDataValue, False),
+        Field("externalAccountIDs", data_list(ExternalID), False),
+    ]
+
+    def __init__(
+        self,
+        name: Optional[str],
+        id: Optional[str],
+        createdAt: Optional[str],
+        type: Optional[str],
+        userRoleOnAccount: Optional[str],
+        externalAccountIDs: Optional[ExternalID],
+    ):
+        self.name = name
+        self.id = id
+        self.createAt = createdAt
+        self.type = type
+        self.userRoleOnAccount = userRoleOnAccount
+        self.externalAccountIDs = externalAccountIDs
+
+
+class ActiveMachine(DataObject):
+    fields = [
+        Field("activeMachines", IntDataValue, False),
+        Field("hours", IntDataValue, False),
+        Field("maximumTrackingReached", BoolDataValue, False),
+    ]
+
+    def __init_(
+        self,
+        activeMachines: Optional[int],
+        hours: Optional[int],
+        maximumTrackingReached: Optional[bool],
+    ):
+        self.activeMachines = activeMachines
+        self.hours = hours
+        self.maximumTrackingReached = maximumTrackingReached
+
+
+class ContractItemReason(str, enum.Enum):
+    SF_ASSET_CREATED = "sf_asset_created"
+    PURCHASE_MADE = "purchase_made"
+    SUBSCRIPTION_RENEWED = "subscription_renewed"
+    REPLACEMENT = "replacement"
+    CONTRACT_CREATED = "contract_created"
+    MIGRATED_ITEM = "migrated_item"
+    EXTEND_CONTRACT_BY_RENEWAL = "extend_contract_by_renewal"
+    TRIAL_STARTED = "trial_started"
+    MANUALLY_CREATED = "manually_created"
+    COMPENSATION = "compensation"
+
+
+class AllowanceInfo(DataObject):
+    fields = [
+        Field("metric", StringDataValue, False),
+        Field("value", IntDataValue, False),
+    ]
+
+    def __init__(self, metric: Optional[str], value: Optional[str]):
+        self.metric = metric
+        self.value = value
+
+
+class Price(DataObject):
+    fields = [
+        Field("value", IntDataValue, False),
+        Field("currency", StringDataValue, False),
+        Field("months", IntDataValue, False),
+    ]
+
+    def __int__(
+        self,
+        value: Optional[int],
+        currency: Optional[str],
+        months: Optional[int],
+    ):
+        self.value = value
+        self.currency = currency
+        self.months = months
+
+
+class RenewalItem(DataObject):
+    fields = [
+        Field("allowance", AllowanceInfo, False),
```

Comment:
Interesting, how does the enum/str interact with everything? does it work appropriately with from_dict and to_dict? We should add unit tests with this kind of usage before relying on it

Reply:
@orndorffgrant I have added some code on this and tests, please check it out.